viewers/pydicom_PIL.py

```diff
@@ -63,45 +69,149 @@ def get_PIL_image(dataset):
         raise TypeError("Cannot show image -- DICOM dataset does not have "
                         "pixel data")
-    # can only apply LUT if these window info exists
-    if ('WindowWidth' not in dataset) or ('WindowCenter' not in dataset):
-        bits = dataset.BitsAllocated
-        samples = dataset.SamplesPerPixel
-        if bits == 8 and samples == 1:
-            mode = "L"
-        elif bits == 8 and samples == 3:
-            mode = "RGB"
-        elif bits == 16:
-            # not sure about this -- PIL source says is 'experimental'
-            # and no documentation. Also, should bytes swap depending
-            # on endian of file and system??
-            mode = "I;16"
-        else:
-            raise TypeError("Don't know PIL mode for %d BitsAllocated "
-                            "and %d SamplesPerPixel" % (bits, samples))
-
-        # PIL size = (width, height)
-        size = (dataset.Columns, dataset.Rows)
-
-        # Recommended to specify all details
-        # by http://www.pythonware.com/library/pil/handbook/image.htm
-        im = PIL.Image.frombuffer(mode, size, dataset.PixelData,
-                                  "raw", mode, 0, 1)
+    
+    bits = dataset.BitsAllocated
+
+    samples = dataset.SamplesPerPixel
+
+    if bits == 8 and samples == 1:
+
+        mode = "L"
 
+    elif bits == 8 and samples == 3:
+
+        mode = "RGB"
+
+    elif bits == 16:
+        # not sure about this -- PIL source says is 'experimental'
+        # and no documentation. Also, should bytes swap depending
+        # on endian of file and system??
+        mode = "I;16"
     else:
-        ew = dataset['WindowWidth']
-        ec = dataset['WindowCenter']
-        ww = int(ew.value[0] if ew.VM > 1 else ew.value)
-        wc = int(ec.value[0] if ec.VM > 1 else ec.value)
-        image = get_LUT_value(dataset.pixel_array, ww, wc)
-        # Convert mode to L since LUT has only 256 values:
-        #   http://www.pythonware.com/library/pil/handbook/image.htm
-        im = PIL.Image.fromarray(image).convert('L')
+        
+        raise TypeError("Don't know PIL mode for %d BitsAllocated "
+                        "and %d SamplesPerPixel" % (bits, samples))
+
+    #LUTification returns ndarrays
+    #can only apply LUT if pydicom is installed
+    image = get_LUT_value(dataset)
 
+    try:
+        #if pixel data has only one frame
+        im = Image.fromarray(image).convert(mode)
+
+    except:
+        #When pixel data has multiple frames, output the first one
+        im = Image.fromarray(image[0]).convert(mode)
+
+    if not brightness_factor == 1.0:
+        #enhancer will enhance
+        enhancer = ImageEnhance.Brightness(im)
+
+        #save enhanced version
+        better_im = enhancer.enhance(brightness_factor)
+
+        return better_im
+
+    
+    #return default brightness
     return im
 
 
 def show_PIL(dataset):
     """Display an image using the Python Imaging Library (PIL)"""
```

Comment:
Shouldn't this also have a brightness factor to pass through to `get_PIL_image`?